src/Internal/OperationsManager.php

```diff
@@ -146,6 +146,8 @@ public function checkTimeoutsAndRetry(TcpPackageConnection $connection): void
                 $this->logDebug('retrying, old corrId %s, operation %s', $oldCorrId, $operation);
                 $this->scheduleOperation($operation, $connection);
             }
+
+            $this->retryPendingOperations = [];
```

Comment:
@pkruithof This is the bad boy for your problem! :)

Reply:
I can confirm that my example works with this. I'll test if my real app also works with it (we run it inside a Kubernetes environment with a HA setup)